### PR TITLE
Drop down border radius

### LINKED ISSUE #6299: border-radius does not apply on components using dropdown lists
## Description  
$border-radius should be applied when set through the theming engine, like:

Snippet:
```
@import '~igniteui-angular/lib/core/styles/themes/index';

$custom-drop-down-theme: igx-drop-down-theme(
    $border-radius: 8px,
);

:host {
    ::ng-deep {
        @include igx-drop-down($custom-drop-down-theme);
        @include igx-button($custom-button-theme);
    }
}
```
 * igniteui-angular version: NA
 * browser: NA

## Steps to reproduce  

1. Navigate and inspect sample

https://stackblitz.com/edit/angular-ddl-border-radius?file=src%2Fapp%2Fdata-entries%2Fdropdown%2Fdropdown-styling%2Fdropdown-styling.component.scss

## Result  
There is no border-radius applied.

## Expected result  
$border-radius being applied through the theming engine.

## Attachments  
API:
https://www.infragistics.com/products/ignite-ui-angular/docs/sass/latest/index.html#mixin-igx-drop-down

### PR DESCRIPTION
**Pull request by desig9stein**
Closes #6299 

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [ ] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/drop-down/_drop-down-component.scss [hunk 1]
```diff
@@ -6,37 +6,21 @@
...
-
 /// @access public
 @include b(igx-drop-down) {
     $this: bem--selector-to-string(&);
     @include register-component(str-slice($this, 2, -1));
 
-    @include _igx-dd-wrapper();
-
-    @extend %igx-drop-down !optional;
+    @include e(list, $m: select) {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/drop-down/_drop-down-component.scss]: Why does this need to be declared here? I.e. why did you move it?

File: projects/igniteui-angular/src/lib/core/styles/components/drop-down/_drop-down-theme.scss [hunk 2]
```diff
@@ -212,21 +212,21 @@
 
     %igx-drop-down-wrapper {
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/drop-down/_drop-down-theme.scss]: Where does this get extended? You have deleted all references of it from the component styles.

**Comment by StefanIvanov** [comment]
Do we have any progress on the two failing test here?
